src/authenticatedFetch/headers/HeadersUtils.ts

```diff
@@ -67,34 +28,23 @@ function hasRawGetter(
  * @param headersToFlatten A structure containing headers potentially in several formats
  */
 export function flattenHeaders(
-  headersToFlatten: Headers | string[][] | Record<string, string> | undefined
+  headersToFlatten: Headers | Record<string, string> | string[][] | undefined
 ): Record<string, string> {
-  if (headersToFlatten === undefined) {
+  if (typeof headersToFlatten === "undefined") {
     return {};
   }
   const flatHeaders: Record<string, string> = {};
-  if (!hasKeys(headersToFlatten)) {
-    if (hasRawGetter(headersToFlatten)) {
-      // This is needed because the headers object passed by Comunica do not
-      // align with either `node-fetch::Headers` or `lib.dom.d.ts::Headers`,
-      // and gets mangled if passed as is to cross-fetch.
-      // eslint-disable-next-line
-      // @ts-ignore
-      return headersToFlatten.raw();
-    } else {
-      return headersToFlatten as Record<string, string>;
-    }
-  } else {
-    // headersToFlatten.keys() SHOULD be valid as per the Headers spec. This seems to be an issue
-    // in lib.dom.d.ts that will need some investigation and potentially a contrib upstream.
-    // eslint-disable-next-line
+
+  // If the headers are already a Record<string, string>,
+  // they can directly be returned.
+  if (typeof headersToFlatten.forEach === "undefined") {
```

Comment:
In general (I've seen this a couple of times in this library), you might want to explicitly check for the desired type. So in this case `typeof headersToFlatten.forEach !== "function"`.

Reply:
Resolved in 66114c1a61050c4e88d093cd04c73eb24b7f434e